collective/pfg/signup/tests/test_adapter.py

```diff
@@ -23,10 +23,135 @@ def test_creation(self):
+        form.invokeFactory('FormStringField', 'username')
+        form.invokeFactory('FormStringField', 'email')
+        form.invokeFactory('FormStringField', 'department')
+        form.mailer.setRecipient_name('Mail Dummy')
+        form.mailer.setRecipient_email('mdummy@address.com')
+        form.signup.full_name_field = 'fullname'
+        form.signup.username_field = 'username'
+        form.signup.email_field = 'email'
+
+    def test_auto_register(self):
+        signup = self.form.signup
+        fields = self.form._getFieldObjects()
+        request = self.LoadRequestForm(fullname='test name', username='test1', email='test@mail.com')
+        signup.onSuccess(fields, request)
+        user = api.user.get(username='test1')
+        self.assertIsNotNone(user, 'Fail to auto-register user')
+
+    def test_add_to_group(self):
+        self.form.signup.user_group_template = Expression('string:staff')
+        signup = self.form.signup
+        fields = self.form._getFieldObjects()
+        request = self.LoadRequestForm(fullname='test name', username='test1', email='test@mail.com', )
+        signup.onSuccess(fields, request)
+        groups = api.group.get_groups(username='test1')
+        self.assertTrue('staff' in [group.id for group in groups],'User not created with staff group')
+
+    def test_add_to_dynamic_group(self):
+        self.form.signup.user_group_template = Expression('string:${department}')
+        signup = self.form.signup
+        fields = self.form._getFieldObjects()
+        request = self.LoadRequestForm(fullname='test name', username='test1', email='test@mail.com',department='IT' )
+        signup.onSuccess(fields, request)
+        groups = api.group.get_groups(username='test1')
+        self.assertTrue('IT' in [group.id for group in groups],'User not created with IT group')
+
+    def test_invalid_domain_name_validation(self):
+        self.form.signup.user_group_template = Expression('string:staff')
+        self.form.signup.email_domain_verification = True
+        signup = self.form.signup
+        fields = self.form._getFieldObjects()
+        request = self.LoadRequestForm(fullname='test name', username='test1', email='test@mail.com')
+        signup.onSuccess(fields, request)
+        user = api.user.get(username='test1')
+        self.assertIsNone(user, 'Fail to validate invalid user email')
+
+    def test_valid_domain_name_validation(self):
+        self.form.signup.user_group_template = Expression('string:staff')
+        self.form.signup.email_domain_verification = True
+        signup = self.form.signup
+        fields = self.form._getFieldObjects()
+        request = self.LoadRequestForm(fullname='test name', username='test2', email='test@plone.org')
+        signup.onSuccess(fields, request)
+        user = api.user.get(username='test2')
+        self.assertIsNotNone(user, 'Fail to validate valid user email')
+
+    def test_approve_user(self):
```

Comment:
we also need a browser test for this form as I can't see it in the UAT site so something is not working

Reply:
browser test or robot test but doesn't need to reproduce all the same test cases. Just show you can see the settings for and the approval form and maybe action them

Reply:
Browser test added